### PR TITLE
Organization and adding token declarations

### PR DESCRIPTION
**Pull request by ehildenb**
The Haskell backend needs `token` declared for the sorts which have `token` in them, even in the semantics module.

This also moves the syntax module declarations, and removes intermediate modules which aren't needed.

### CODE REVIEW

File: data.md [hunk 1]
```diff
@@ -78,6 +18,18 @@ module WASM-DATA
     imports BYTES
 ```
 
+### WASM Token Sorts
```
Comment (ttuegel) [inline data.md]: I'm a little confused by this: The `syntax` declarations already had `token` attributes, so why was this change necessary? This almost seems like a frontend bug to me.

File: README.md [hunk 2]
```diff
@@ -93,12 +93,6 @@ After installing the above dependencies, make sure the submodules are setup:
 git submodule update --init --recursive
 ```
 
-Similarly, you'll need to setup K's Rust dependencies:
-
-```sh
-./deps/k/llvm-backend/src/main/native/llvm-backend/install-rust
-```
-
 If you haven't already setup K's OCaml dependencies more recently than February 1, 2019, then you also need to setup the K OCaml dependencies:
```
Comment (hjorthjort) [inline README.md]: While we're at it, is this still relevant (February 1, 2019 part)?

File: data.md (hunk 1 above)
Comment (ehildenb) [reply data.md]: The module used for verification is `KWASM-LEMMAS`, which doesn't have `WASM-TOKEN-SYNTAX` in its import chain. We only ever import `WASM-TOKEN-SYNTAX` when trying to parse programs.

Comment (ttuegel) [reply data.md]: That's strange, because I was also getting errors due to the missing `token` attribute when trying to run concrete programs.

File: README.md (hunk 2 above)
Comment (ehildenb) [reply README.md]: That was the date I wrote this text. Still kinda relevant, though we've switched to LLVM backend as default. But you still need to setup ocaml dependencies if you want to use the OCaml backend.

File: data.md (hunk 1 above)
Comment (ehildenb) [reply data.md]: I believe this should be resolved now.
